sisl/viz/plotly/plots/pdos.py

```diff
@@ -287,7 +296,12 @@ def _read_TBtrans(self, root_fdf, tbt_nc):
         self.PDOS = tbt_sile.DOS(sum=False).data.T
         self.E = tbt_sile.E
 
-        self.geometry = tbt_sile.read_geometry().sub(tbt_sile.a_dev)
+        try:
+            self.geometry = self.get_sile("root_fdf").read_geometry(output=True)
+        except (FileNotFoundError, TypeError):
+            self.geometry = tbt_sile.read_geometry()
+
+        self.geometry = self.geometry.sub(tbt_sile.a_dev)
```

Comment:
Hmm.. I am not sure doing the `sub` is always a good idea. The problem is that some users may expect to work on full geometry indices and not a subset. Also, some of the tbt_nc sile stuff returns data in the *global* picture i.e. with buffer atoms etc. Why are you subbing?

Reply:
Ok, this is just because in the tests that I did `tbt_sile.DOS` returns only data for orbitals in the scattering region. Are there cases where the DOS is returned for all the atoms? (I wrote this entry point quickly because I wanted to use the PdosPlot for my first tbtrans PDOS calculation, I didn't know all the different possibilities, and I still don't)

> The problem is that some users may expect to work on full geometry indices and not a subset.

Yes, I agree. I'll have to think of some way to handle the mapping from "full geometry" to "only scattering" indices (or maybe this is already in `tbt_sile`?). Although this has not bothered me at all because I'm always using categories (or dicts) to select the atoms. Knowing this, I'm not sure if this is a high priority thing that should be fixed in this PR or not, what do you think?

Reply:
This is one of the parts where sisl is not really consistent... :(

Some of the quantities, DOS returns for the device atoms only.
Other quantiities such as sparse matrices (bond-current and density matrices) are returned in the full geometry.

I initially did the DOS, thinking that users would always only think about the device region.
However, when I did the sparse matrices it seemed more natural to return everything, since that is the matrices from siesta anyways. Perhaps I should make them only return the device region....

Reply:
I don't know what to say, I have no opinion on what would be best :sweat_smile: